db.go

```diff
@@ -1166,13 +1166,26 @@ func (db *DB) allocate(txid common.Txid, count int) (*common.Page, error) {
 	p.SetId(db.rwtx.meta.Pgid())
 	var minsz = int((p.Id()+common.Pgid(count))+1) * db.pageSize
 	if minsz >= db.datasz {
-		if err := db.mmap(minsz); err != nil {
-			if err == berrors.ErrMaxSizeReached {
-				return nil, err
+		if !db.readOnly && db.MaxSize > 0 {
```

Comment:
a readonly TXN will never call `allocate`
```suggestion
		if db.MaxSize > 0 {
```